backends-common/postgres/src/main/java/org/apache/james/backends/postgres/utils/JamesPostgresConnectionFactory.java

```diff
@@ -21,9 +21,11 @@
 
 import org.apache.james.core.Domain;
 
-import io.r2dbc.spi.Connection;
+import io.r2dbc.postgresql.api.PostgresqlConnection;
 import reactor.core.publisher.Mono;
 
 public interface JamesPostgresConnectionFactory {
-    Mono<? extends Connection> getConnection(Domain domain);
+    Mono<? extends PostgresqlConnection> getConnection(Domain domain);
+
+    Mono<? extends PostgresqlConnection> getConnection();
```

Comment:
How about:
```java
    Mono<? extends PostgresqlConnection> getConnection(Optional<Domain> domain);
```

- Upon domain existence (virtualhosting on): we would return a mapped connection for that domain
- Upon domain non-existence (virtualhosting off or for queries not tied to domain scope e.g. init tables upon James startup): we would return default connection.